c7n/resources/quotas.py

```diff
@@ -54,29 +54,41 @@ def augment(self, resources):
         retry = get_retry(('TooManyRequestsException',))
 
         def get_quotas(client, s):
-            quotas = {}
-            token = None
-            kwargs = {
-                'ServiceCode': s['ServiceCode'],
-                'MaxResults': self.batch_size
+            def _get_quotas(client, s, attr):
+                quotas = {}
+                token = None
+                kwargs = {
+                    'ServiceCode': s['ServiceCode'],
+                    'MaxResults': self.batch_size
+                }
+
+                while True:
+                    if token:
+                        kwargs['NextToken'] = token
+                    response = retry(
+                        getattr(client, attr),
+                        **kwargs
+                    )
+                    rquotas = {q['QuotaCode']: q for q in response['Quotas']}
+                    token = response.get('NextToken')
+                    new = set(rquotas) - set(quotas)
+                    quotas.update(rquotas)
+                    if token is None:
+                        break
+                    # ssm, ec2, kms have bad behaviors.
+                    elif token and not new:
+                        break
+                return quotas.values()
+
+            dquotas = {
+                q['QuotaCode']: q
+                for q in _get_quotas(client, s, 'list_aws_default_service_quotas')
             }
-            while True:
-                if token:
-                    kwargs['NextToken'] = token
-                response = retry(
-                    client.list_service_quotas,
-                    **kwargs
-                )
-                rquotas = {q['QuotaCode']: q for q in response['Quotas']}
-                token = response.get('NextToken')
-                new = set(rquotas) - set(quotas)
-                quotas.update(rquotas)
-                if token is None:
-                    break
-                # ssm, ec2, kms have bad behaviors.
-                elif token and not new:
-                    break
-
+            quotas = {
+                q['QuotaCode']: q
+                for q in _get_quotas(client, s, 'list_service_quotas')
+            }
+            quotas.update(dquotas)
```

Comment:
don't you want the other way around? ie use update defaults with actuals?

Reply:
It looks like the quotas are completely different, for example for s3:

```
$ aws service-quotas list-aws-default-service-quotas --service-code s3 --query "Quotas[*].QuotaName"
[
    "Access Points",
    "Bucket policy",
    "Bucket tags",
    "Buckets",
    "CRR rules",
    "Event notifications",
    "Lifecycle rules",
    "Maximum part size",
    "Minimum part size",
    "Multi-Region Access Point Regions",
    "Multi-Region Access Points",
    "Object size",
    "Object size (Console upload)",
    "Object tags",
    "Parts",
    "Replication transfer rate",
    "S3 Glacier: Number of random restore requests.",
    "S3 Glacier: Provisioned capacity units"
]
```

```
$ aws service-quotas list-service-quotas --service-code s3 --query "Quotas[*].QuotaName"
[]
```

Reply:
@thisisshi 
Please reconsider using the other way around. This morning I received alerts from almost all my accounts. 🤣 
It uses the default value rather than the applied quota value. See below for an example. The `Quota` is a default one, the applied one is 1000+.
```
ServiceName: Amazon Elastic Compute Cloud (Amazon EC2)
QuotaName: Running On-Demand Standard (A, C, D, H, I, M, R, T, Z) instances
Quota: 5.0
Usage: 325.0
```